pkg/hyper/helper.go

```diff
@@ -300,21 +301,30 @@ func getCpuLimitFromCgroup(cgroupParent string) (int32, error) {
 	if err != nil {
 		return -1, err
 	}
-	contents, err := ioutil.ReadFile(filepath.Join(mntPath, cgroupParent, cpuQuotaCgroupFile))
+	cgroupPath := filepath.Join(mntPath, cgroupParent)
+	cpuQuota, err := readCgroupFileToInt32(cgroupPath, cpuQuotaCgroupFile)
+	if err != nil {
+		return -1, err
+	}
+	cpuPeriod, err := readCgroupFileToInt32(cgroupPath, cpuPeriodCgroupFile)
 	if err != nil {
 		return -1, err
 	}
 
-	cpuQuotaStr := strings.TrimSpace(string(contents))
+	// HyperContainer only support int32 vcpu number, and we need to use `math.Ceil` to make sure vcpu number is always enough.
+	vcpuNumber := float64(cpuQuota) / float64(cpuPeriod)
+	return int32(math.Ceil(vcpuNumber)), nil
+}
 
-	if value, err := strconv.Atoi(cpuQuotaStr); err == nil {
-		// k8s use 100 milicpu to describe cpu quota
-		vcpuNumber := value / cpu100m
-		// but hyper only support int32 vcpu number, should be one vcpu at least
-		if vcpuNumber < 1 {
-			vcpuNumber = 1
-		}
-		return int32(vcpuNumber), nil
+// readCgroupFileToInt32 reads contents from given `cgroupPath/cgroupFile` and returns its int32 value
+func readCgroupFileToInt32(cgroupPath, cgroupFile string) (int32, error) {
+	contents, err := ioutil.ReadFile(filepath.Join(cgroupPath, cgroupFile))
+	if err != nil {
+		return -1, err
+	}
+	strValue := strings.TrimSpace(string(contents))
+	if value, err := strconv.Atoi(strValue); err == nil {
+		return int32(value), nil
```

Comment:
I think we should read int64 from cgroup files (int32 can only hold about 2GB memory) and use `ParseInt` instead of `Atoi` in case loosing accuracy.